src/gui/src/PatternEditor/PatternEditor.cpp

```diff
@@ -409,6 +410,60 @@ void PatternEditor::selectInstrumentNotes( int nInstrument )
 	m_selection.updateWidgetGroup();
 }
 
+
+///
+/// Align selected (or all) notes to the current grid
+///
+void PatternEditor::alignToGrid() {
+
+	// Align selected notes to grid.
+	if ( m_pPattern == nullptr || m_nSelectedPatternNumber == -1 ) {
+		// No pattern selected.
+		return;
+	}
+
+	Hydrogen *pHydrogen = Hydrogen::get_instance();
+	auto pInstrumentList = pHydrogen->getSong()->getInstrumentList();
+	QUndoStack *pUndo = HydrogenApp::get_instance()->m_pUndoStack;
+	validateSelection();
+
+	std::list< Note * > notesToAlign;
+	for ( auto pNote : m_selection ) {
+		notesToAlign.push_back( pNote );
+	}
+	// If selection is empty, use all notes in the pattern
+	if ( notesToAlign.empty() ) {
```

Comment:
I was a little bit surprised by this behavior. Mostly, because the action is listed next a lot of selection-based actions, like Cut, Copy, Select All, Clear Selection.

Possible side effects: When working on a pattern, shrinking it to e.g. half its size, and pressing <kbd>Ctrl</kbd> + <kbd>A</kbd> only notes currently accessible within the reduced pattern length will be selected. When using `get_notes()` all notes regardless whether they are accessible or not will be selected.